framework/src/DriverFramework.cpp

```diff
@@ -629,21 +629,10 @@ void HRTWorkQueue::process(void)
 
 		// pthread_cond_timedwait uses absolute time
 		ts = offsetTimeToAbsoluteTime(now + next);
+		DF_LOG_DEBUG("HRTWorkQueue::process waiting for work (%" PRIi64 ")", diff);
+		// Wait until next expiry or until a new item is rescheduled
+		pthread_cond_timedwait(&g_reschedule_cond, &g_hrt_lock, &ts);
```

Comment:
`0` or `ETIMEDOUT` are the success cases (at least according to the comment above the call to `pthread_cond_timedwait()`), so my thought was just to handle the case in which the return value is something *other* than those 2 options, at least to log an error.